src/components/DownloadLimitEntry.vue

```diff
@@ -129,13 +130,15 @@ export default defineComponent({
 	},
 
 	methods: {
+		n,
+		t,
+
 		handleUpdateLimit(limit: string) {
 			this.limit = Number(limit) // emitted <input> value is string so we parse it to number
 		},
 
 		async onSave() {
-			const isValid = typeof this.limit === 'number' && this.limit > 0
-			if (!isValid) {
+			if (typeof this.limit !== 'number' || this.limit <= 0) {
```

Comment:
With that helper Typescript can not type check the `limit` property, so directly using it in the `if` allows Typescript to narrow down the type further down the function :)